src/topic_table_model.cpp

```diff
@@ -793,8 +793,10 @@ void TopicTableModel::parseAggregated(const DDS::DynamicData_var& data, const st
             continue;
         }
 
-	std::string scoped_member_name = namePrefix.empty() ? md->name() : namePrefix + "." + md->name();
-        const OpenDDS::XTypes::TypeKind member_tk = OpenDDS::XTypes::get_base_type(md->type())->get_kind();
+        std::string scoped_member_name = namePrefix.empty() ? md->name() : namePrefix + "." + md->name();
+        const DDS::DynamicType_var type = md->type();
+        const DDS::DynamicType_var base_type = OpenDDS::XTypes::get_base_type(type.in());
```

Comment:
```suggestion
        const DDS::DynamicType_var base_type = OpenDDS::XTypes::get_base_type(md->type());
```
This is one of the odd things about IDL Valuetypes.  The "accessors" like `md->type()` do not increase the refcount on the object being returned.  (IDL Unions are the same.)